tools/tracer: fails to build with libc++

## Steps to reproduce
```
$ export CC=clang CXX=clang++ CXXFLAGS="$CXXFLAGS -stdlib=libc++" LDFLAGS="$LDFLAGS -fuse-ld=bfd"
$ cmake -DBUILD_TOOLS=ON -B /tmp/MediaSDK_build -G Ninja
$ cmake --build /tmp/MediaSDK_build
```

## Clang 10
```c++
tools/tracer/dumps/dump.h:175:42: error: dynamic_cast from rvalue to reference type 'std::ostringstream &' (aka 'basic_ostringstream<char> &')
        str += "* " + structName + "=" + ToHexFormatString(_struct) + "\n";
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tracer/dumps/dump.h:81:37: note: expanded from macro 'ToHexFormatString'
    #define ToHexFormatString( x ) (dynamic_cast< std::ostringstream & >( ( std::ostringstream() << std::hex << pVoidToHexString((void*)x) ) ).str() )
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tracer/dumps/dump.h:182:42: error: dynamic_cast from rvalue to reference type 'std::ostringstream &' (aka 'basic_ostringstream<char> &')
        str += "* " + structName + "=" + ToHexFormatString(_struct) + "\n";
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tracer/dumps/dump.h:81:37: note: expanded from macro 'ToHexFormatString'
    #define ToHexFormatString( x ) (dynamic_cast< std::ostringstream & >( ( std::ostringstream() << std::hex << pVoidToHexString((void*)x) ) ).str() )
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tracer/dumps/dump.h:200:56: error: dynamic_cast from rvalue to reference type 'std::ostringstream &' (aka 'basic_ostringstream<char> &')
                    name = structName + ".ExtParam[" + ToString(i) + "]";
                                                       ^~~~~~~~~~~
tools/tracer/dumps/dump.h:70:28: note: expanded from macro 'ToString'
    #define ToString( x )  dynamic_cast< std::ostringstream & >( \
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tracer/dumps/dump.h:490:67: error: dynamic_cast from rvalue to reference type 'std::ostringstream &' (aka 'basic_ostringstream<char> &')
                    str += "WARNING: Can't read from ExtParam[" + ToString(i) + "]!\n";
                                                                  ^~~~~~~~~~~
tools/tracer/dumps/dump.h:70:28: note: expanded from macro 'ToString'
    #define ToString( x )  dynamic_cast< std::ostringstream & >( \
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tracer/dumps/dump.cpp:370:23: error: dynamic_cast from rvalue to reference type 'std::ostringstream &' (aka 'basic_ostringstream<char> &')
        name += "(" + ToString(io) + ")";
                      ^~~~~~~~~~~~
tools/tracer/dumps/dump.h:70:28: note: expanded from macro 'ToString'
    #define ToString( x )  dynamic_cast< std::ostringstream & >( \
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
5 errors generated.
```
## GCC 10
```c++
In file included from tools/tracer/dumps/dump.cpp:31:
tools/tracer/dumps/dump.h: In member function 'std::__1::string DumpContext::dump(std::__1::string, const void*)':
tools/tracer/dumps/dump.h:81:37: error: invalid 'static_cast' from type 'std::__1::basic_ostringstream<char>' to type 'std::__1::ostringstream&' {aka 'std::__1::basic_ostringstream<char>&'}
   81 |     #define ToHexFormatString( x ) (dynamic_cast< std::ostringstream & >( ( std::ostringstream() << std::hex << pVoidToHexString((void*)x) ) ).str() )
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tracer/dumps/dump.h:182:42: note: in expansion of macro 'ToHexFormatString'
  182 |         str += "* " + structName + "=" + ToHexFormatString(_struct) + "\n";
      |                                          ^~~~~~~~~~~~~~~~~
In file included from tools/tracer/dumps/dump.cpp:31:
tools/tracer/dumps/dump.cpp: In function 'std::__1::string GetIOPattern(mfxU32)':
tools/tracer/dumps/dump.h:70:28: error: invalid 'static_cast' from type 'std::__1::basic_ostringstream<char>' to type 'std::__1::ostringstream&' {aka 'std::__1::basic_ostringstream<char>&'}
   70 |     #define ToString( x )  dynamic_cast< std::ostringstream & >( \
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   71 |         ( std::ostringstream() << std::dec << x ) ).str()
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tracer/dumps/dump.cpp:370:23: note: in expansion of macro 'ToString'
  370 |         name += "(" + ToString(io) + ")";
      |                       ^~~~~~~~
```

--- a/tools/tracer/dumps/dump.h
+++ b/tools/tracer/dumps/dump.h
@@ -67,18 +67,19 @@ bool _IsBadReadPtr(void *ptr, size_t size);
 #define DUMP_FIELD_RESERVED(_field) \
     str += structName + "." #_field "[]=" + DUMP_RESERVED_ARRAY(_struct._field) + "\n";
 
-    #define ToString( x )  dynamic_cast< std::ostringstream & >( \
+    #define ToString( x )  static_cast< std::ostringstream const & >( \
         ( std::ostringstream() << std::dec << x ) ).str()
 
-    #define TimeToString( x ) dynamic_cast< std::ostringstream & >( \
+    #define TimeToString( x ) static_cast< std::ostringstream const & >( \
             ( std::ostringstream() << std::left << std::setw(4) << std::dec << x <<" msec") ).str()
 
     /*
-    #define ToHexFormatString( x ) dynamic_cast< std::ostringstream & >( \
+    #define ToHexFormatString( x ) static_cast< std::ostringstream const & >( \
             ( std::ostringstream() << std::hex << x ) ).str()
     */
 
-    #define ToHexFormatString( x ) (dynamic_cast< std::ostringstream & >( ( std::ostringstream() << std::hex << pVoidToHexString((void*)x) ) ).str() )
+    #define ToHexFormatString( x ) (static_cast< std::ostringstream const & >( \
+            ( std::ostringstream() << std::hex << pVoidToHexString((void*)x) ) ).str() )
 /*
 #define DEFINE_GET_TYPE(type) \
     template<> \
@@ -118,7 +119,8 @@ public:
 
     template<typename T>
     inline std::string toString( T x, eDumpFormat format = DUMP_DEC){
-        return dynamic_cast< std::ostringstream & >(( std::ostringstream() << ((format == DUMP_DEC) ? std::dec : std::hex) << x )).str();
+        return static_cast< std::ostringstream const & >(
+            ( std::ostringstream() << ((format == DUMP_DEC) ? std::dec : std::hex) << x )).str();
     }
 
     const char* get_bufferid_str(mfxU32 bufferid);

